File: CIELRCPT.c
#include <stdio.h>
int main()
{
int t,p,i,k,count;
long long int f[12];
f[0]=1;
for(i=1;i<=11;i++)
f[i]=2*f[i-1];
scanf("%d",&t);
for(i=0;i<t;i++)
{
count=0;
scanf("%d",&p);
for(k=0;p/f[k]!=0;)
k++;
for(k=k;k>=0;)
if(p>=f[k])
{
p=p-f[k];
count++;
}
else
k--;
printf("%d\n",count);
}
return 0;
}

File: cielrcpt.py
t=int(input())
while t>0:
    count=0
    p=int(input())
    k=0
    while p>pow(2,k) and k<11:
        k=k+1
    while k>=0:
        q=p//pow(2,k)
        p=p%pow(2,k)
        count+=q
        k=k-1
    print(count)
    t=t-1
